conan/tools/cmake/cmakedeps/templates/target_configuration.py

```diff
@@ -31,7 +31,8 @@ def context(self):
                 "config_suffix": self.config_suffix,
                 "deps_targets_names": ";".join(deps_targets_names),
                 "components_names": components_names,
-                "configuration": self.cmakedeps.configuration}
+                "configuration": self.cmakedeps.configuration,
+                "nosoname": str(self.nosoname).upper()}
```

Comment:
I think I would try to put this as a ``xxxx-data.cmake`` variable, then use it. Like, all the parametrization of a dependency should go to ``xxx-data.cmake``, and the rest of the files use the data from there to act accordingly.